decidim-elections/app/forms/decidim/elections/admin/setup_form.rb

```diff
@@ -58,6 +80,22 @@ def bulletin_board
           @bulletin_board ||= context[:bulletin_board] || Decidim::Elections.bulletin_board
         end
 
+        def needs_census?
+          election.component.participatory_space.respond_to?(:dataset)
```

Comment:
Was there some problem with the approach of checking whether the census is needed at #9369:
https://github.com/decidim/decidim/blob/d89d3ec8ac031cb707414d47c55542c9bbc25edd/decidim-elections/app/forms/decidim/elections/admin/setup_form.rb#L45-L48

I was just thinking the `dataset` attribute name sounds pretty generic, so it might introduce other issues along the line if we assume the `dataset` always means census.

Reply:
Didn't think about this, seems reasonable. Done!
Added a system spec as well!